src/Nox.Lib/Middlewares/RelatedEndpointsMiddleware.cs

```diff
@@ -0,0 +1,141 @@
+﻿using Microsoft.AspNetCore.Http;
+using Nox.Lib;
+using Nox.Solution;
+
+namespace Nox.Middlewares;
+
+internal class RelatedEndpointsMiddleware
+{
+    private readonly PathString _apiPrefix;
+
+    private readonly RequestDelegate _next;
+
+    private readonly IEnumerable<string> _entitiesPluralNames;
+
+    private readonly IEnumerable<string> _navigationNames;
+
+    public RelatedEndpointsMiddleware(RequestDelegate next, NoxSolution solution)
+    {
+        _next = next;
+
+        _apiPrefix = solution.Presentation.ApiConfiguration.ApiRoutePrefix;
+
+        if (solution.Domain is null)
+        {
+            _entitiesPluralNames = Array.Empty<string>();
+            _navigationNames = Array.Empty<string>();
+        }
+        else
+        {
+            _entitiesPluralNames = solution.Domain.Entities.Select(e => e.PluralName).ToList();
+            _navigationNames = solution.Domain.Entities
+                .SelectMany(e => e.Relationships.Select(r => e.GetNavigationPropertyName(r)))
+                .Distinct()
+                .ToList();
+        }
+    }
+
+    public async Task InvokeAsync(HttpContext context)
+    {
+        var requestPath = context.Request.Path;
+
+        if (!requestPath.HasValue)
+        {
+            await _next(context);
+            return;
+        }
+
+        if (!requestPath.StartsWithSegments(_apiPrefix))
+        {
+            await _next(context);
+            return;
+        }
+
+        if (context.Request.Method == "PATCH")
+        {
+            var segments = ParsePath(requestPath);
+
+            if (!ValidateSegments(segments))
+            {
+                await _next(context);
+                return;
+            }
+
+            var newPath = BuildNewPatchPath(segments);
+            context.Request.Path = newPath;
+            context.Request.QueryString = context.Request.QueryString;
+        }
+
+        await _next(context);
+        return;
+    }
+
+
+    private List<string> ParsePath(string path)
+    {
+        List<string> segments = new List<string>();
+
+        if (path.StartsWith(_apiPrefix, StringComparison.OrdinalIgnoreCase))
+        {
+            path = path.Substring(_apiPrefix.Value!.Length);
+        }
+
+        if (path.StartsWith("/", StringComparison.Ordinal)) 
```

Comment:
it always start by /... do we need the comparation?

Reply:
updated